modules/mill-codegen-plugin/test/src/smithy4s/codegen/mill/Smithy4sModuleSpec.scala

```diff
@@ -0,0 +1,122 @@
+package smithy4s.codegen.mill
+
+import mill.testkit.MillTestKit
+import mill.scalalib._
+import mill._
+import munit.Location
+import sourcecode.FullName
+
+class Smithy4sModuleSpec extends munit.FunSuite {
+  object testKit extends MillTestKit
+
+  val coreDep =
+    ivy"com.disneystreaming.smithy4s::smithy4s-core:${smithy4s.codegen.BuildInfo.version}"
+
+  object foo extends testKit.BaseModule with Smithy4sModule {
+    override def scalaVersion = "2.13.8"
+    override def ivyDeps = Agg(coreDep)
+  }
+
+  object bar extends testKit.BaseModule with Smithy4sModule {
+    override def moduleDeps = Seq(foo)
+    override def scalaVersion = "2.13.8"
+    override def ivyDeps = Agg(coreDep)
+  }
+
+  test("codegen runs") {
+    val ev = testKit.staticTestEvaluator(foo)(FullName("codegen-runs"))
+    os.write.over(
+      foo.millSourcePath / "smithy" / "foo.smithy",
+      s"""|$$version: "2"
+          |
+          |namespace basic
+          |
+          |string MyNewString""".stripMargin,
+      createFolders = true
+    )
+
+    compileWorks(foo, ev)
+    checkFileExist(
+      ev.outPath / "smithy4sOutputDir.dest" / "scala" / "basic" / "MyNewString.scala",
+      shouldExist = true
+    )
+  }
+
+  test("multi-module codegen works") {
+    val fooEv = testKit.staticTestEvaluator(foo)(FullName("multi-module-foo"))
+    val barEv = testKit.staticTestEvaluator(foo)(FullName("multi-module-bar"))
```

Comment:
> I think this is more a criticism of https://github.com/com-lihaoyi/mill/pull/1893/files rather than this PR though 

UX can be improved as we learn more by using.